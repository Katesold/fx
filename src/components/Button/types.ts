export interface StyledButtonProps {
    size: string;
    color?: string;
    onClick: () => {},
    disabled?: boolean,
}