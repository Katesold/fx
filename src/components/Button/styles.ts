import styled from 'styled-components';
import { StyledButtonProps } from './types';

export const StyledButton = styled.button<StyledButtonProps>`
    background: ${props => (props.color === 'green' ? '#098375' : 'blue')};
    border-radius: 4px;
    border: 2px solid #098375;
    color: #FFF;
    margin: 0 20px 0 auto;
    padding: 1em 2em;
    font-size: ${props => (props.size === 'small' ? '12px' : '16px')};
    font-weight: 600;
    cursor: pointer;
    transition: box-shadow 0.2s ease-in-out;
    width: 120px;
    &:hover {
            border-color: #000000;
            transition: border-color 0.2s ease-in-out;
            box-shadow: 0 3px 7px rgb(0 0 0 / 30%);
          }
          &:focus {
            border-color: #000000;
            box-shadow: 0 3px 7px rgb(0 0 0 / 30%);
            transition: border-color 0.2s ease-in-out;
          }
          &:active {
            border-color: #000000;
            box-shadow: 0 5px 15px rgba(0,0,0,0.3);
            transition: border-color 0.2s ease-in-out;
          }
`