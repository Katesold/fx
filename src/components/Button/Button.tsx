import { StyledButton } from './styles';

export const Button = ({ text = 'Submit', disabled = false, ...rest} : { text?: string, disabled?: boolean, onClick: any}) => {
    return (
        <StyledButton disabled={disabled} size="small" color="green" {...rest} >{text}</StyledButton>
    )
};