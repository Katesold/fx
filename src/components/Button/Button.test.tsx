import { Button } from './Button';
import { screen, render } from '@testing-library/react';

describe('Button components are being tested', () => {
    const props = {
        onClick: () => {},
        size: 'small',
        color: 'green',
        disabled: false,
    };
    it('should match snapshot', () => {
        const { container } = render(<Button {...props} />);
        expect(container).toMatchSnapshot();
    });
    it('should display the button correctly with props', () => {
        render(<Button {...props} />);
        const button = screen.getByText('Submit');
        expect(button).toBeInTheDocument();
    });
    it('should display the disabled button correctly with props', () => {
        const newProps = {
            ...props,
            disabled: true,
        };
        const { container } = render(<Button {...newProps} />);
        expect(container).toMatchSnapshot();
    });
});