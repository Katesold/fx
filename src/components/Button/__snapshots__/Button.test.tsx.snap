// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components are being tested should display the disabled button correctly with props 1`] = `
<div>
  <button
    class="sc-bcXHqe eXfJSU"
    color="green"
    disabled=""
  >
    Submit
  </button>
</div>
`;

exports[`Button components are being tested should match snapshot 1`] = `
<div>
  <button
    class="sc-bcXHqe eXfJSU"
    color="green"
  >
    Submit
  </button>
</div>
`;
